site/app/templates/grading/electronic/Details.twig

```diff
@@ -1,4 +1,20 @@
 {% import _self as self %} {# Required to use macros in same template #}
+{#
+creates an array of submissions so that they can all be passed into the javascript and then the submission controller for regrading
+#}
+{% set submission = [] %}
+{% for section in sections %}
+    {% for row in section.rows %}
+        {% if row.graded_gradeable.getAutoGradedGradeable().getActiveVersion() > 0 %}
+            {% if is_team %}
+                {% set submission = submission | merge([{gradeable_id: row.graded_gradeable.gradeable.getId, submitter: row.graded_gradeable.getSubmitter().getTeam().getMemberUsers()[0].getId(), highest_version: row.graded_gradeable.getAutoGradedGradeable().getHighestVersion(), active_version: row.graded_gradeable.getAutoGradedGradeable().getActiveVersion() }]) %}
+            {% else %}
+                {% set submission = submission | merge([{gradeable_id: row.graded_gradeable.gradeable.getId, submitter: row.graded_gradeable.getSubmitter().getUser().getId(), highest_version: row.graded_gradeable.getAutoGradedGradeable().getHighestVersion(), active_version: row.graded_gradeable.getAutoGradedGradeable().getActiveVersion() }]) %}
+            {% endif %}
+        {% endif %}
+    {% endfor %}
+{% endfor %}
+<input type="hidden" id="submission" value="{{ submission|json_encode }}">
```

Comment:
Why render all of the submissions to the user? This would also happen every time that page is loaded which isn't that efficient. Couldn't this be generated in the SubmissionController?